### ISSUE TITLE
Reason behind using optimizer.get_lr() over optimizer.get_last_lr()? 

### ISSUE THREAD

**Issue by Priahi** [HarukiYqM/Non-Local-Sparse-Attention#2]
Hi there, 

I have been able to reproduce both your experiments and have read your paper as well and I do not understand why you do not use the last learning rate from the previous epoch, ie through get_last_lr(), in the start of the train step, but instead you opt for get_lr(), which returns a value that is only scaled by some gamma factor (implementation of this function is here for your reference: https://github.com/pytorch/pytorch/blob/fde94e75568b527b424b108c272793e096e8e471/torch/optim/lr_scheduler.py#L344-L352). 

Associated Pytorch warning is the following: lib/python3.8/site-packages/torch/optim/lr_scheduler.py:416: UserWarning: To get the last learning rate computed by the scheduler, please use `get_last_lr()`.
  warnings.warn("To get the last learning rate computed by the scheduler, "

Looking forward to hearing back from you soon and all the best,

Parsa Riahi

**Comment by HarukiYqM** [comment]
Hi, thanks pointing this out. In fact, this is a legacy problem of EDSR codebase https://github.com/sanghyun-son/EDSR-PyTorch. I didn’t modify it since it does not affect training. Please feel free to make a pull request to fix it.

**Comment by Priahi** [comment]
Sure thanks I will make that change. 
In the meantime, is there any way I can pause my train experiment and start at an arbitrary checkpoint, I know you have the --resume flag, but this doesn't affect the learning rate or the number of epochs (which seem to simply follow the args). Does this all have to be set manually? If so, are these three parameters the only relevant ones or are there more?

Best, 

Parsa Riahi

**Comment by HarukiYqM** [comment]
Hi, Yes, it needs to be set manually and only these three parameters are relevant.

**Comment by Priahi** [comment]
Thanks!
